### PR TITLE
Python: added num_records parameter to text memory skill

### PR DESCRIPTION
**Pull request by eavanvalkenburg**
### Motivation and Context

The `text_memory_skill` in the core skills was setup to only return 1 record, while the underlying memories all have support for multiple, so I added a parameter to the recall function to use that.

### Description

Added parameter, passed that on to the memory call.
### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/core_skills/text_memory_skill.py [hunk 1]
```diff
@@ -39,10 +46,11 @@ async def recall_async(self, ask: str, context: SKContext) -> str:
         Args:
             ask -- The question to ask the memory
             context -- Contains the 'collection' to search for information
-                and the 'relevance' score to use when searching
+                , the 'relevance' score to use when searching
+                and the 'num_records' to retrieve.
 
         Returns:
-            The nearest item from the memory store
+            The nearest item from the memory store as a comma-separated string or empty string if not found.
```
Comment (shawncal) [inline python/semantic_kernel/core_skills/text_memory_skill.py:53]: Not sure comma-separated is useful here, as the memories themselves can contain punctuation. In the dotnet flavor, we accept a "limit" param and return the results as a json collection:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Skills/Skills.Core/TextMemorySkill.cs

There's a good opportunity to align the languages here (parameter naming and output expectation)

Comment (eavanvalkenburg) [reply python/semantic_kernel/core_skills/text_memory_skill.py:53]: This also adds the limit parameter, but still returns a single string, that is comma-separated when multiple results are returned (aka when limit > 1 and there are more then 1 results that match the threshold)

Comment (eavanvalkenburg) [reply python/semantic_kernel/core_skills/text_memory_skill.py:53]: ah sorry, I see that the naming is different I will update to get it inline with the dotnet version

File: python/semantic_kernel/core_skills/text_memory_skill.py [hunk 2]
```diff
@@ -39,10 +47,11 @@ async def recall_async(self, ask: str, context: SKContext) -> str:
         Args:
             ask -- The question to ask the memory
             context -- Contains the 'collection' to search for information
-                and the 'relevance' score to use when searching
+                , the 'relevance' score to use when searching
+                and the 'limit' to retrieve.
```
Comment (TaoChenOSU) [inline python/semantic_kernel/core_skills/text_memory_skill.py:51]: nit: 'limit' of relevant memories to retrieve.

**Comment by eavanvalkenburg** [comment]
I already solved that!

**Review by alexchaomander — APPROVED** [review]

**Pull request merged**